### ISSUE TITLE
[analytics] Add event tracking for LuckyOrange

### ISSUE THREAD

**Issue by jabby09** [KlimaDAO/klimadao#1086]
# Carbonmark LuckyOrange Events

## Docs 
https://developers.luckyorange.com/libraries/browser
If using browser implementation use the ES5 syntax when calling ready() since we're using GTM

## Events to track

Login
`LO.events.track('Login')`

UserID
`LO.visitor.identify('wallet', { user: 'username', name: 'Display Name' })`

Purchase: Buy Clicked
`LO.events.track('Purchase: Buy Clicked')`

Purchase: Continue Clicked
`LO.events.track('Purchase: Continue Clicked')`

Purchase: Approve Clicked
`LO.events.track('Purchase: Approve Clicked')`

Purchase: Submit Clicked
`LO.events.track('Purchase: Submit Clicked')`

Purchase: Purchase Completed
`LO.events.track('Purchase: Purchase Completed')`

Purchase: Purchase Modal Closed
`LO.events.track('Purchase: Purchase Modal Closed')`

Retire: Retire Button Clicked
`LO.events.track('Retire: Retire Button Clicked')`

Listing: Sell Clicked
`LO.events.track('Listing: Sell Clicked')`

Listing: Approve Clicked
`LO.events.track('Listing: Approve Clicked')`

Listing: Listing Created
`LO.events.track('Listing: Listing Created')`

Listing: Edit Clicked
`LO.events.track('Listing: Edit Clicked')`

Listing: Updated
`LO.events.track('Listing: Updated')`

Listing: Deleted
`LO.events.track('Listing: Deleted')`

**Comment by jabby09** [comment]
This is perhaps a good for @biwano but we need to check on his availability.

**Comment by biwano** [comment]
@jabby09 Do you know if we already have an account and  SITE_ID for Lucky Orange?

**Comment by jabby09** [comment]
> @jabby09 Do you know if we already have an account and SITE_ID for Lucky Orange?

@biwano  yes, site ID is 96a45e35

<img width="1325" alt="Screenshot 2023-04-29 at 10 29 18 AM" src="https://user-images.githubusercontent.com/103535195/235305467-930d1388-dfd4-46a1-8d07-c1e16878791b.png">

**Comment by biwano** [comment]
Some notes:
 - Login is triggered each time a page is loaded. Maybe this is no the desired behaviour?
 - Should we differentiate how we track "buying" for pools and for user listings?

**Comment by jabby09** [comment]
@biwano 

> Should we differentiate how we track "buying" for pools and for user listings?

I would think yes
We will need the differentiation on retire as well once all that functionality is implemented.

**Comment by jabby09** [comment]
@biwano 

> Login is triggered each time a page is loaded. Maybe this is no the desired behaviour?

I think the intent here is to track how many time the LOGIN button is clicked. I think what you are saying is that it is possible to not log out, close your browser, and return to the site and still be logged in and that would trigger the login event. Is that correct?

@sy-zygy what specific behaviour do you want to track here?

**Comment by jabby09** [comment]
First PR related to this issue #1115 

LuckyOrange Events 

- [x] Login
LO.events.track('Login')
Note: Tracks only identified Carbonmark users

- [x] UserID
LO.visitor.identify('wallet', { user: 'username', name: 'Display Name' })
Note: Tracks only identified Carbonmark  users

- [x] Purchase: Buy Clicked
LO.events.track('Purchase: Buy Clicked')

- [x] Purchase: Continue Clicked
LO.events.track('Purchase: Continue Clicked')
Note: Triggers only if the form is valid.

- [ ] Purchase: Approve Clicked
LO.events.track('Purchase: Approve Clicked')

- [ ] Purchase: Submit Clicked
LO.events.track('Purchase: Submit Clicked')

- [ ] Purchase: Purchase Completed
LO.events.track('Purchase: Purchase Completed')

- [ ] Purchase: Purchase Modal Closed
LO.events.track('Purchase: Purchase Modal Closed')

- [ ] Retire: Retire Button Clicked
LO.events.track('Retire: Retire Button Clicked')

- [ ] Listing: Sell Clicked
LO.events.track('Listing: Sell Clicked')

- [ ] Listing: Approve Clicked
LO.events.track('Listing: Approve Clicked')

- [ ] Listing: Listing Created
LO.events.track('Listing: Listing Created')

- [ ] Listing: Edit Clicked
LO.events.track('Listing: Edit Clicked')

- [ ] Listing: Updated
LO.events.track('Listing: Updated')

- [ ] Listing: Deleted
LO.events.track('Listing: Deleted')

**Comment by biwano** [comment]
> hi @biwano looking at the data in lucky orange and some feedback/questions...
> 
> * Seeing two custom api login events, one lowercase and one uppercase - is this a bug?
Definitely a bug, but it is weird. There is only one version in the released code. Either some events where sent from my development machine or it is something done automatically by the LuckyOrange framework.
> 
> <img alt="Screenshot 2023-05-10 at 10 11 54 AM" width="592" src="https://user-images.githubusercontent.com/103535195/237404605-545e1c3c-cd65-4dd9-b801-023c4a592ddb.png">
> * The user ID event `LO.visitor.identify('wallet', { user: 'username', name: 'Display Name' })` the ID is coming over as "wallet" but the intention is that this should be the 0x wallet address, not the word wallet. So for example the data might look more like this `LO.visitor.identify('0x0000...', { user: 'jabby', name: 'jabby' })`
>   ![Screenshot 2023-05-10 at 10 18 15 AM](https://user-images.githubusercontent.com/103535195/237405123-c7559655-cd03-4d65-b1dc-a86be002aeae.png)

Makes perfect sense now :)

> * Seeing an instance of the `Retire: Retire Button Clicked` event but do not see it checked off in your list of events that got released
>   ![Screenshot 2023-05-10 at 10 23 52 AM](https://user-images.githubusercontent.com/103535195/237405787-f0cf2242-1ab0-4a1a-8a0e-1835ecda85c3.png)

Yes my bad. Checklist updated.

> * Are you sure `LO.events.track('Login')` event is firing every time the user logs in. It seemed to work for me yesterday when I logged in but today I'm not seeing the custom event trigger when I login?

It should. I will check again on a local production build. Maybe there is a difference in behaviour with the production code.

> * @sy-zygy and I are still looking at the data in LO and trouble shooting a few things but it does look like the custom events are firing and the data is showing up. Can you please do a 2nd PR to fix the bugs listed above and to release the rest of the events.

On it.

**Comment by biwano** [comment]
Hi @jabby09 and @sy-zygy 

PR #1138 is ready. Please check my notes, specialty for the way login is tracked now.

@Atmosfearful Can you check my changes to Web3Modal? I am bit intrusive but I did not find a lighter way to detect consistently if the login was manual or from cache.

**Comment by jabby09** [comment]
Hi @biwano generally the events seem to be working.

Still need a few adjustments on the User ID piece (apologies is the requirements were not clear).

Here is what is currently showing in LO...

![Image](https://github.com/KlimaDAO/klimadao/assets/103535195/4780ef44-b8d3-43a3-adcd-c4a68234f529)

Based on this `LO.visitor.identify('wallet', { user: 'username', name: 'Display Name' })`

The field `ID` should contain the wallet 0x address.
The field `user` should contain the Carbonmark profile @handle name
There needs to be a field called `name` which should contain the Carbonmark profile display name
The field `wallet` is not required since the wallet address is stored in ID

![Image](https://github.com/KlimaDAO/klimadao/assets/103535195/249ea31b-87ac-4eec-a298-c73f73303cf2)

Make sense?

**Comment by biwano** [comment]
Hi @jabby09,

No problem. I put up a new PR. The code should reflect the requirements now.
A few notes:
- The **name** atttribute has a special spot in the interface. See capture below: 
![image](https://github.com/KlimaDAO/klimadao/assets/11857992/88598c70-5ce3-4ba4-b797-cb458f61316d)

- I removed the wallet attribute and populated the ID. But note that I did not find a way yet to remove an attribute. So for people previously identified, the wallet attribute still shows. I can make further research if it is really important.

- Just to share knowledge: During my tests I used the "Delete User" feature from the LuckyOrange interface on my personal profile to simulate an **unknown** user from LuckyOrange point of view. It completely broke tracking from my browser. I had to manually delete LuckyOrange related cookies. Use with caution.
